test/test_framework.py

```diff
@@ -275,8 +316,11 @@ def _on_qux(self, event: ops.EventBase, extra: typing.Optional[typing.Any] = Non
             framework.observe(pub.baz, obs._on_baz)  # type: ignore
         framework.observe(pub.qux, obs._on_qux)
 
-    def test_on_pre_commit_emitted(self):
-        framework = self.create_framework(tmpdir=self.tmpdir)
+    def test_on_pre_commit_emitted(self, request: pytest.FixtureRequest):
+        tmpdir = Path(tempfile.mkdtemp())
+        request.addfinalizer(lambda: shutil.rmtree(tmpdir))
```

Comment:
Sorry for being dense, but I still don't understand why we can't use a fixture to create (and remove) this directory? At least for the simple cases like this (and others below).